osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -48,7 +53,18 @@ public RulesetStore(IDatabaseContextFactory factory)
         /// </summary>
         public IEnumerable<RulesetInfo> AvailableRulesets { get; private set; }
 
-        private Assembly resolveRulesetAssembly(object sender, ResolveEventArgs args) => loadedAssemblies.Keys.FirstOrDefault(a => a.FullName == args.Name);
+        private Assembly resolveRulesetDependencyAssembly(object sender, ResolveEventArgs args)
+        {
+            var asm = new AssemblyName(args.Name);
+
+            // the requesting assembly may be located out of the executable's base directory, thus requiring manual resolving of its dependencies.
+            // this assumes the only explicit dependency of the ruleset is the game core assembly.
+            // the ruleset dependency on the game core assembly requires manual resolving, transient dependencies should be resolved automatically
```

Comment:
I think this means to say
```suggestion
            // the ruleset dependency on the game core assembly requires manual resolving, transitive dependencies should be resolved automatically
```